tests/unit/test_exporter.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python3
+"""Test exporter daemon."""
+import asyncio
+import unittest.mock as mock
+
+import pytest
+
+
+class TestExporterDaemon:
+    """Exporter daemon test class."""
+
+    def test_fixture(self, exporter_daemon):
+        """See if the helper fixture works to load charm configs."""
+        stats_exporter_daemon = exporter_daemon()
+        assert stats_exporter_daemon is not None
+
+    def test_setup_logging(self, exporter_daemon):
+        """Test setup logger."""
+        statsd = exporter_daemon(True)
+        statsd.logger.setLevel.assert_called_with("DEBUG")
+
+    def test_parse_config(self, exporter_daemon):
+        """Test config parsing."""
+        statsd = exporter_daemon()
+
+        assert isinstance(statsd.config["exporter"]["port"].get(), int)
+        assert isinstance(statsd.config["exporter"]["collect_interval"].get(), int)
+
+        assert statsd.config["exporter"]["port"].get() == 9748
+        assert statsd.config["exporter"]["collect_interval"].get() == 15
+
+    @pytest.mark.asyncio
+    async def test_trigger(self, exporter_daemon):
+        """Test trigger function."""
+        statsd = exporter_daemon()
+        await statsd.trigger(once=True)
+
+        statsd.logger.info.assert_called_with(
+            "Gauges collected and ready for exporting."
+        )
```

Comment:
Validate the logger message may be too much. It creates the difficulty during refactor and may violate this rule 
> Don’t Couple Your Tests With Implementation Details

(This can be discussed)